src/irmin-pack/pack.ml

```diff
@@ -276,8 +278,10 @@ struct
 
     let find t k =
       Lwt_mutex.with_lock t.pack.lock (fun () ->
-          let v = unsafe_find t k in
-          Lwt.return v)
+          try
+            let v = unsafe_find t k in
+            Lwt.return v
+          with exn -> Lwt.fail exn)
```

Comment:
Why do we need an extra `try .. with` here?

Reply:
we don't :)